main/src/library/Result.flix

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Result {
+
+    ///
+    /// Returns `true` iff `r` is `Ok(v)`.
+    ///
+    def isOk[a, b](r: Result[a, b]): Bool = match r with {
+        case Ok(v) => true
+        case Err(e) => false
+    }
+
+    ///
+    /// Returns `true` iff `r` is `Err(e)`.
+    ///
+    def isErr[a, b](r: Result[a, b]): Bool = match r with {
+        case Ok(v) => false
+        case Err(e) => true
+    }
+
+    ///
+    /// Returns `v` if `r` is `Ok(v)`. Undefined semantics if `r` is `Err(e)`.
+    ///
+    def ok[a, b](r: Result[a, b]): a = match r with {
```

Comment:
I see why you added these, but often they are not provided. It is a bit strange considering that Option *does* provide `get`.

Actually neither are necessary. You can write:

```
let Some(x) = o
let Ok(x) = r
```
etc.

I propose that we remove `ok` and `err` (but leave `get` on option for historic purposes)